ssh/ssh.go

```diff
@@ -169,20 +173,39 @@ func DecryptSSHKey(key []byte, password []byte) (gossh.Signer, error) {
 	return gossh.NewSignerFromKey(sshkey)
 }
 
-func resolveSigner(priv privateKey) (gossh.Signer, error) {
-	signer, err := gossh.ParsePrivateKey(priv.body)
-	if err != nil && strings.Contains(err.Error(), "cannot decode encrypted private keys") {
-		fmt.Fprintf(os.Stderr, "This SSH key is encrypted. Please enter passphrase for key '%s':", priv.path)
-		var passphrase []byte
-		passphrase, err = terminal.ReadPassword(int(syscall.Stdin))
+func resolveAuthMethod(priv privateKey) (gossh.AuthMethod, error) {
+	var authmethod gossh.AuthMethod
```

Comment:
Currently this method will return a `nil` authmethod. 

Here is the fix (I tested it):

```go
	if len(priv.body) > 0 {
		signer, err := gossh.ParsePrivateKey(priv.body)
		if err != nil && strings.Contains(err.Error(), "cannot decode encrypted private keys") {
			fmt.Fprintf(os.Stderr, "This SSH key is encrypted. Please enter passphrase for key '%s':", priv.path)
			var passphrase []byte
			passphrase, err = terminal.ReadPassword(int(syscall.Stdin))
			if err != nil {
				return nil, err
			}
			fmt.Fprintln(os.Stderr)
			signer, err = DecryptSSHKey(priv.body, passphrase)
		}

		return gossh.PublicKeys(signer), err
	}

	sshAuthSock := os.Getenv("SSH_AUTH_SOCK")
	if len(sshAuthSock) == 0 {
		return nil, fmt.Errorf("No key provided and no SSH_AUTH_SOCK env variable set, unable to resolve auth")
	}

	agentUnixSock, err := net.Dial("unix", sshAuthSock)
	if err != nil {
		return nil, err
	}

	return gossh.PublicKeysCallback(agent.NewClient(agentUnixSock).Signers), nil
```